tools/report-converter/codechecker_report_converter/sparse/output_parser.py

```diff
@@ -0,0 +1,91 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+
+import logging
+import os
+import re
+
+from ..output_parser import Message, Event, BaseParser
+
+LOG = logging.getLogger('ReportConverter')
+
+
+class SparserParser(BaseParser):
+    """ Parser for sparse console output. """
+
+    def __init__(self):
+        super(SparserParser, self).__init__()
+
+        # Regex for parsing a sparse message.
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':' and a space.
+            r'(?P<column>\d+?): '
+            # Severity followed by a ':'.
+            r'(?P<severity>(error|warning)):'
+            # Checker message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+        # Match a sparse detailed info
+        self.message_detailed_info = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':' , 4 spaces (not a tab because a 
+            # tab has 7 spaces)
+            r'(?P<column>\d+?):    '
+            # Detailed info type (expected | got) followed by a ':'.
+            r'(?P<severity>(expected|got|struct)) '
+            # Checker message
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+    def parse_message(self, it, line):
+        """Parse the given line.
+
+        Returns a (message, next_line) pair or throws a StopIteration.
+        The message could be None.
+        """
+        match = self.message_line_re.match(line)
+        if match is None:
+            return None, next(it)
+        
+        message_body = match.group('message').strip()
+
+        message = Message(
+            os.path.abspath(match.group('path')),
+            int(match.group('line')),
+            int(match.group('column')),
+            message_body,
+            "")
+
+        try:
+            line = next(it)
+            detailed_match = self.message_detailed_info.match(line)
+            while detailed_match is not None:
+                message_body = message_body + " \n" \
+                            + detailed_match.group('severity').strip() + " " \
+                            + detailed_match.group('message').strip()
+                line = next(it)
+                detailed_match = self.message_detailed_info.match(line)
+                message = Message(
```

Comment:
This is not a good idea to create a separate report from the details. I recommend you to create notes here instead of a new message here because these informations are connected to the report created above.

For more information check how we parse  UBSAN results:
https://github.com/Ericsson/codechecker/blob/5e4a4f5d39045694a6ff46cd3ac3ba63d4b41d13/tools/report-converter/codechecker_report_converter/sanitizers/ub/output_parser.py#L58-L82